client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -163,7 +163,7 @@ var CheckoutThankYou = React.createClass( {
 	 * @returns {*[]} an array of varying size with the component instance, then an optional purchase object possibly followed by a domain name
 	 */
 	getComponentAndPrimaryPurchaseAndDomain: function() {
-		if ( this.isDataLoaded() ) {
+		if ( this.isDataLoaded() && this.props.receiptId ) {
```

Comment:
I wonder if it would be better to test for the presence of the list of purchases instead in all these checks (here and above) as conditions and clauses seem disconnected at first glance. To be more specific, when I see that condition I wonder why we have to check if a receipt id is available to process a list of purchases. Then I remember that's because we retrieve the list of purchases from the receipt but that's not obvious. The `isDataLoaded()` function doesn't really help to make things clearer since it also checks `this.props.receiptId`:

```javascript
isDataLoaded: function() {
  return ! this.props.receiptId || this.props.receipt.hasLoadedFromServer;
},
```

It's true though that `props.receipt.data.purchases` would make conditions a bit long. Maybe we could update the `mapStateToProps()` function to extract the list of purchases and pass it as `props.purchases` to the component?